service-providers/ip-packet-router/src/lib.rs

```diff
@@ -235,26 +253,34 @@ impl IpPacketRouter {
     ) -> Result<(), IpPacketRouterError> {
         log::info!("Received message: {:?}", reconstructed.sender_tag);
 
-        let headers = etherparse::SlicedPacket::from_ip(&reconstructed.message).map_err(|err| {
-            log::warn!("Received non-IP packet: {err}");
-            IpPacketRouterError::PacketParseFailed { source: err }
-        })?;
-
-        let (src_addr, dst_addr): (IpAddr, IpAddr) = match headers.ip {
-            Some(etherparse::InternetSlice::Ipv4(ipv4_header, _)) => (
-                ipv4_header.source_addr().into(),
-                ipv4_header.destination_addr().into(),
-            ),
-            Some(etherparse::InternetSlice::Ipv6(ipv6_header, _)) => (
-                ipv6_header.source_addr().into(),
-                ipv6_header.destination_addr().into(),
-            ),
-            None => {
-                log::warn!("Received non-IP packet");
-                return Err(IpPacketRouterError::PacketMissingHeader);
+        // We don't forward packets that we are not able to parse. BUT, there might be a good
+        // reason to still forward them.
+        //
+        // For example, if we are running in a mode where we are only supposed to forward
+        // packets to a specific destination, we might want to forward them anyway.
+        //
+        // TODO: look into this
+        let (packet_type, src_addr, dst_addr, dst) = parse_packet(&reconstructed.message)?;
+
+        let dst_str = dst.map_or(dst_addr.to_string(), |dst| dst.to_string());
+        log::info!("Received packet: {packet_type}: {src_addr} -> {dst_str}");
+
+        // Filter check
+        if let Some(dst) = dst {
+            if !self.request_filter.check_address(&dst).await {
+                let log_msg = format!("Failed filter check: {dst:?}");
+                log::warn!("{log_msg}");
+
+                // TODO: send back a response here
```

Comment:
well, do we? isn't it the case that if the underlying client fails to get acks multiple in a row, it fail just give up? like say when you're trying to load a website in your browser with disabled internet, it will just give you no internet error at some point

Reply:
Yep, but I was thinking that optionally we could be "polite" and send back a request that is parseable in in the nym-client to log that this was the reason there are no acks. Not sure, just an idea